### PR TITLE
auto-11214: migrate more tests to foundry

### PR DESCRIPTION
**Pull request by FelixFan1992**

### CODE REVIEW

**Review by ibrajer — DISMISSED** [review]

File: contracts/src/v0.8/automation/test/v2_3/BaseTest.t.sol [hunk 1]
```diff
@@ -356,40 +355,44 @@ contract BaseTest is Test {
...
-      gasLimits[i] = registry.getUpkeep(ids[i]).performGas;
-      performDatas[i] = new bytes(0);
-      uint8 triggerType = registry.getTriggerType(ids[i]);
-      if (triggerType == 0) {
-        triggers[i] = _encodeConditionalTrigger(
-          AutoBase.ConditionalTrigger(uint32(block.number - 1), blockhash(block.number - 1))
-        );
-      } else {
-        revert("not implemented");
+  function _transmit(uint256[] memory ids, Registry registry, bytes4 selector) internal {
```
Comment (RyanRHall) [inline contracts/src/v0.8/automation/test/v2_3/BaseTest.t.sol]: It was a bit confusing for me why we're passing `bytes4(0)` everywhere in the tests. I would suggest doing something like this...

```solidity
// internal functions, these can be used by AutomationRegistry.t.sol

// tests single upkeep, expects success
function _transmit(uint256 id, Registry registry) internal {
  uint256[] memory ids = new uint256[](1);
  ids[0] = id;
  _handleTransmit(ids, registry, bytes4(0));
}

// tests multiple upkeeps, expects success
function _transmit(uint256[] memory ids, Registry registry) internal {
  _handleTransmit(ids, registry, bytes4(0));
}

// tests single upkeep, expects revert
function _transmitAndExpectRevert(uint256 id, Registry registry, bytes4 selector) internal {
  uint256[] memory ids = new uint256[](1);
  ids[0] = id;
  _handleTransmit(ids, registry, selector);
} 

// private function not exposed to actual testing contract

function _handleTransmit(uint256[] memory ids, Registry registry, bytes4 selector) private {
  // same logic from the current _transmit() function
}

```

File: contracts/src/v0.8/automation/test/v2_3/AutomationRegistry2_3.t.sol [hunk 2]
```diff
@@ -1977,3 +1982,789 @@ contract MigrateReceive is SetUp {
...
+      address(TARGET1),
+      config.maxPerformGas,
+      UPKEEP_ADMIN,
+      uint8(Trigger.CONDITION),
+      address(linkToken),
+      "",
+      "",
+      ""
+    );
+  }
```
Comment (RyanRHall) [inline contracts/src/v0.8/automation/test/v2_3/AutomationRegistry2_3.t.sol]: I would call this a `registerUpkeep()` test, not a `pause()` test

File: contracts/src/v0.8/automation/test/v2_3/AutomationRegistry2_3.t.sol [hunk 3]
```diff
@@ -1977,3 +1982,789 @@ contract MigrateReceive is SetUp {
...
+
+  function test_revertsWhen_transmitInPausedRegistry() external {
+    vm.startPrank(registry.owner());
+    registry.pause();
+
+    _transmit(usdUpkeepID18, registry, Registry.RegistryPaused.selector);
+  }
```
Comment (RyanRHall) [inline contracts/src/v0.8/automation/test/v2_3/AutomationRegistry2_3.t.sol]: I would call this a `transmit()` test, not a `pause()` test

**Review by RyanRHall — CHANGES_REQUESTED** [review]
wow great job converting all these @FelixFan1992  💪 🎉

File: contracts/src/v0.8/automation/test/v2_3/AutomationRegistry2_3.t.sol (hunk 2 above)
Comment (FelixFan1992) [reply contracts/src/v0.8/automation/test/v2_3/AutomationRegistry2_3.t.sol]: done

File: contracts/src/v0.8/automation/test/v2_3/AutomationRegistry2_3.t.sol (hunk 3 above)
Comment (FelixFan1992) [reply contracts/src/v0.8/automation/test/v2_3/AutomationRegistry2_3.t.sol]: i will work on a different PR to migrate more tests (more trickier tests) including for `transmit` and i will add it there.

File: contracts/src/v0.8/automation/test/v2_3/BaseTest.t.sol (hunk 1 above)
Comment (FelixFan1992) [reply contracts/src/v0.8/automation/test/v2_3/BaseTest.t.sol]: done

**Review by RyanRHall — APPROVED** [review]
LGTM 👍

**Review by cmalec — APPROVED** [review]

**Pull request merged**